temporalcli/commands.operator_namespace_test.go

```diff
@@ -0,0 +1,204 @@
+		"--address", s.Address(),
+		nsName,
+	)
+	s.NoError(res.Err)
+
+	res = s.Execute(
+		"operator", "namespace", "list",
+		"--address", s.Address(),
+		"--output", "json",
+	)
+	s.NoError(res.Err)
+	var listResp []workflowservice.DescribeNamespaceResponse
+	s.NoError(json.Unmarshal(res.Stdout.Bytes(), &listResp))
+	var uuid string
+	for _, ns := range listResp {
+		if ns.NamespaceInfo.Name == nsName {
+			uuid = ns.NamespaceInfo.Id
+		}
+	}
+	s.NotEmpty(uuid)
+
+	res = s.Execute(
+		"operator", "namespace", "describe",
+		"--address", s.Address(),
+		"--output", "json",
+		nsName,
+	)
+	s.NoError(res.Err)
+	var describeResp workflowservice.DescribeNamespaceResponse
+	s.NoError(temporalcli.UnmarshalProtoJSONWithOptions(res.Stdout.Bytes(), &describeResp, true))
+	s.Equal(nsName, describeResp.NamespaceInfo.Name)
+
+	// Validate default values
+	s.Equal("active", describeResp.ReplicationConfig.ActiveClusterName)
+	s.Equal("active", describeResp.ReplicationConfig.Clusters[0].ClusterName)
+	s.Len(describeResp.NamespaceInfo.Data, 0)
+	s.Len(describeResp.NamespaceInfo.Description, 0)
+	s.Len(describeResp.NamespaceInfo.OwnerEmail, 0)
+	s.Equal(false, describeResp.IsGlobalNamespace)
+	s.Equal(enums.ARCHIVAL_STATE_DISABLED, describeResp.Config.HistoryArchivalState)
+	s.Len(describeResp.Config.HistoryArchivalUri, 0)
+	s.Equal(describeResp.Config.WorkflowExecutionRetentionTtl.AsDuration(), 72*time.Hour)
+	s.Equal(enums.ARCHIVAL_STATE_DISABLED, describeResp.Config.VisibilityArchivalState)
+	s.Len(describeResp.Config.VisibilityArchivalUri, 0)
+}
+
+func (s *SharedServerSuite) TestNamespaceUpdate_Verbose() {
+	nsName := "test-namespace-update-verbose"
+
+	res := s.Execute(
+		"operator", "namespace", "create",
+		"--address", s.Address(),
+		"--description", "description before",
+		"--email", "email@before",
+		"--retention", "1",
+		nsName,
+	)
+	s.NoError(res.Err)
+
+	res = s.Execute(
+		"operator", "namespace", "update",
+		"--address", s.Address(),
+		"--description", "description after",
+		"--email", "email@after",
+		"--retention", "2",
+		"--output", "json",
+		"--verbose",
+		nsName,
+	)
+	s.NoError(res.Err)
+	s.ContainsOnSameLine(res.Stdout.String(), "workflowExecutionRetentionTtl", "172800s")
+	s.ContainsOnSameLine(res.Stdout.String(), "description", "description after")
+	s.ContainsOnSameLine(res.Stdout.String(), "ownerEmail", "email@after")
+
+	res = s.Execute(
+		"operator", "namespace", "describe",
+		"--address", s.Address(),
+		"--output", "json",
+		nsName,
+	)
+	s.NoError(res.Err)
+	var describeResp workflowservice.DescribeNamespaceResponse
+	s.NoError(temporalcli.UnmarshalProtoJSONWithOptions(res.Stdout.Bytes(), &describeResp, true))
+
+	s.Equal("description after", describeResp.NamespaceInfo.Description)
+	s.Equal("email@after", describeResp.NamespaceInfo.OwnerEmail)
+	s.Equal(48*time.Hour, describeResp.Config.WorkflowExecutionRetentionTtl.AsDuration())
+}
+
+func (s *SharedServerSuite) TestNamespaceUpdate_Data() {
```

Comment:
Wonder if you want to move this with your create-list-and-describe so you don't mess with the `default` namespace everyone else uses. Can move delete over there too. I understand it can seem a bit wrong to test so many separate things in a single test, but in these cases of integration with several parts it makes sense.

Reply:
Right. I moved this to TestNamespaceUpdate.